components/SubmissionForm.vue

```diff
@@ -168,4 +188,46 @@ const validateFields = (e: Event) => {
     }
     submissionStore.submit()
 }
+
+const submissionInputFields = [
+    { field: 'location', validate: validations.validateGoogleMapsUrlInput, validationReference: isValidGoogleMapsUrl },
+    { field: 'lastName', validate: validations.validateUserSubmittedLastName, validationReference: isValidLastName },
+    { field: 'firstName', validate: validations.validateUserSubmittedFirstName, validationReference: isValidFirstName },
+    { field: 'selectLanguage1', validate: validations.validateFirstSpokenLanguage, validationReference: isValidPrimarySpokenLanguage },
+    { field: 'selectLanguage2', validate: validations.validateSecondSpokenLanguage, validationReference: isValidSecondarySpokenLanguage }
+]
+
+const initialValidationCheck =
+async (inputValue: string, inputValidationCheck: (inputValue: string) => boolean, field: ValidationReference) => {
+    validationCheckedPreviously.value = true
+    await nextTick()
+    const validationReference = useCorrectValidation(field)
+    validationReference.value = inputValidationCheck(inputValue)
+}
+
+const useCorrectValidation = (field: ValidationReference) => {
+    switch (field) {
+        case ValidationReference.Location:
+            return isValidGoogleMapsUrl
+        case ValidationReference.LastName:
+            return isValidLastName
+        case ValidationReference.FirstName:
+            return isValidFirstName
+        case ValidationReference.PrimarySpokenLanguage:
+            return isValidPrimarySpokenLanguage
+        case ValidationReference.SecondarySpokenLanguage:
+            return isValidSecondarySpokenLanguage
+    }
+}
+
+submissionInputFields.forEach(({ field, validate, validationReference }) => {
```

Comment:
```suggestion
submissionInputFields.forEach(({ field, validationFunction, validationReference }) => {
```

Reply:
Not using this anymore. I have 5 watches since I think it is more readable. It isn't super dry. But like before. These fields are pretty static so the necessity for dynamic fields is maybe an overcomplicated approach.